tests/src/test/scala/org/apache/openwhisk/common/etcd/EtcdWorkerTests.scala

```diff
@@ -0,0 +1,176 @@
+  val instanceId = schedulerId
+
+  behavior of "EtcdWorker"
+
+  it should "elect leader and send completion ack to parent" in {
+    val mockEtcd = mock[EtcdClient]
+
+    val key = "testKey"
+    val value = "testValue"
+    val leader = Right(EtcdLeader(key, value, leaseId))
+    val etcdWorker = TestActorRef(EtcdWorker.props(mockEtcd, leaseService.ref), self)
+
+    (mockEtcd
+      .electLeader(_: String, _: String, _: Lease))
+      .expects(key, value, *)
+      .returns(Future.successful(leader))
+
+    etcdWorker ! ElectLeader(key, value, recipient = self)
+
+    expectMsg(ElectionResult(leader))
+    expectMsg(FinishWork(key))
+  }
+
+  it should "register initial data when doesn't exit and send completion ack to parent" in {
+    val mockEtcd = mock[EtcdClient]
+
+    val key = "testKey"
+    val value = "testValue"
+    val etcdWorker = TestActorRef(EtcdWorker.props(mockEtcd, leaseService.ref), self)
+
+    (mockEtcd
+      .putTxn(_: String, _: String, _: Long, _: Long))
+      .expects(key, value, *, *)
+      .returns(Future.successful(TxnResponse.newBuilder().setSucceeded(true).build()))
+
+    etcdWorker ! RegisterInitialData(key, value, recipient = Some(self))
+
+    expectMsg(FinishWork(key))
+    expectMsg(InitialDataStorageResults(key, Right(Done())))
+  }
+
+  it should "attempt to register initial data when exists and send completion ack to parent" in {
+    val mockEtcd = mock[EtcdClient]
+
+    val key = "testKey"
+    val value = "testValue"
+    val etcdWorker = TestActorRef(EtcdWorker.props(mockEtcd, leaseService.ref), self)
+
+    (mockEtcd
+      .putTxn(_: String, _: String, _: Long, _: Long))
+      .expects(key, value, *, *)
+      .returns(Future.successful(TxnResponse.newBuilder().setSucceeded(false).build()))
+
+    etcdWorker ! RegisterInitialData(key, value, recipient = Some(self))
+
+    expectMsg(FinishWork(key))
+    expectMsg(InitialDataStorageResults(key, Left(AlreadyExist())))
+  }
+
+  it should "register data and send completion ack to parent" in {
+    val mockEtcd = mock[EtcdClient]
+
+    val key = "testKey"
+    val value = "testValue"
+    val etcdWorker = TestActorRef(EtcdWorker.props(mockEtcd, leaseService.ref), self)
+
+    (mockEtcd
+      .put(_: String, _: String, _: Long))
+      .expects(key, value, leaseId)
+      .returns(Future.successful(PutResponse.newBuilder().build()))
+
+    etcdWorker ! RegisterData(key, value)
+
+    expectMsg(FinishWork(key))
+  }
+
+  it should "delete data when watcher closed" in {
+    val mockEtcd = mock[EtcdClient]
+
+    val key = "testKey"
+    val etcdWorker = TestActorRef(EtcdWorker.props(mockEtcd, leaseService.ref), self)
+
+    (mockEtcd
+      .del(_: String))
+      .expects(key)
+      .returns(Future.successful(DeleteRangeResponse.newBuilder().build()))
+
+    etcdWorker ! WatcherClosed(key, false)
+
+    expectMsg(FinishWork(key))
+  }
+
+  it should "retry request after failure if lease does not exist" in {
```

Comment:
Can we also add other negative tests for `RegisterData` and `RegisterInitialData`?